policy-controller/k8s/index/src/outbound/index/egress_network.rs

```diff
@@ -61,14 +61,20 @@ impl EgressNetwork {
 // Attempts to find the best matching network for a certain discovery look-up.
 // Logic is:
 // 1. if there are Egress networks in the source_namespace, only these are considered
+// 2. otherwise only networks from the global egress network namespace are considered
 // 2. the target IP is matched against the networks of the EgressNetwork
 // 3. ambiguity is resolved as by comparing the networks using compare_matched_egress_network
 pub(crate) fn resolve_egress_network<'n>(
     addr: IpAddr,
     source_namespace: String,
+    global_external_network_namespace: Arc<String>,
```

Comment:
would it simplify things to just take a `&str` here since we don't actually need to take ownership of the String?